ArtistList/app/src/main/java/com/grzesica/przemek/artistlist/Model/DataFetcher.java

```diff
@@ -0,0 +1,218 @@
+import com.grzesica.przemek.artistlist.Container.IDataFetcherDIBuilder;
+import com.grzesica.przemek.artistlist.Container.IHttpHandlerDIBuilder;
+import com.grzesica.przemek.artistlist.Container.IJsonObjectExtended;
+import com.grzesica.przemek.artistlist.Viewer.ArtistListActivity;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.util.concurrent.AbstractExecutorService;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+
+import static android.content.ContentValues.TAG;
+
+/**
+ * Created by przemek on 05.03.18.
+ */
+
+public class DataFetcher implements IDataFetcher {
+
+    private AbstractExecutorService mThreadPoolExecutor;
+    private Context mContext;
+    private IDataBaseAdapterDIBuilder mDataBaseAdapterDIBuilder;
+    private IHttpHandlerDIBuilder mHttpHandlerDIBuilder;
+    private IJsonObjectExtended mJsonObjectExtended;
+    private final Handler mHandler;
+
+    // Gets the number of available cores
+    private static int NUMBER_OF_CORES = Runtime.getRuntime().availableProcessors();
+    // Sets the amount of time an idle thread waits before terminating
+    private static final long KEEP_ALIVE_TIME = 2000;
+    // Sets the Time Unit to Milliseconds
+    private static final TimeUnit KEEP_ALIVE_TIME_UNIT = TimeUnit.MILLISECONDS;
+
+    public static final String JSON_URL = "http://i.img.co/data/data.json";
+
+    public DataFetcher(IDataFetcherDIBuilder builder, Context context){
+        this.mContext = context;
+        this.mDataBaseAdapterDIBuilder = ((DataFetcherDIBuilder)builder).mDataBaseAdapterDIBuilder;
+        this.mJsonObjectExtended = ((DataFetcherDIBuilder)builder).mJsonObjectExtended;
+        this.mHttpHandlerDIBuilder = ((DataFetcherDIBuilder)builder).mHttpHandlerDIBuilder;
+        this.mHandler = ((DataFetcherDIBuilder)builder).mHandler;
+        this.mThreadPoolExecutor = ((DataFetcherDIBuilder)builder).mThreadPoolExecutor;
+    }
+
+    @Override
+    public void getData(){
+        IHttpHandler httpHandler = ((HttpHandlerDIBuilder)mHttpHandlerDIBuilder)
+                .byteArrayOutputStream()
+                .strBuilder()
+                .extendedUrl()
+                .extendedBufferedReader()
+                .build();
+
+        DataBaseAdapterDIBuilder dataBaseAdapterDIBuilder = (DataBaseAdapterDIBuilder) mDataBaseAdapterDIBuilder;
+        IDataBaseAdapter dataBaseAdapter = dataBaseAdapterDIBuilder
+                .contentValues()
+                .dataBaseHelperDIBuilder()
+                .build(mContext);
+
+        String jsonStr = httpHandler.jsonServiceCall(JSON_URL);
+
+        if (jsonStr != null) {
+            try {
+                JSONObject jsonObj = (JSONObject) mJsonObjectExtended.setJsonObject(jsonStr);
+                // Getting JSON Array node
+                JSONArray artistJsonArray = jsonObj.getJSONArray("artists");
+                ((DataBaseAdapter)dataBaseAdapter).open(1, true);
```

Comment:
hmm, tego to szczerze nie rozumiem. Wydaje mi sie ze przekombinowales z obsluga DB. Wystarczy przeciez 1 baze podniesc i pobrac do niej uchwyt. Uchwyt ten mozesz uzywac wszedzie gdzie checesz... baze zamykasz np. przy wyjsciu zaplikacji.